frontend/packages/noobaa-storage-plugin/src/components/status-card/statuses.tsx

```diff
@@ -1,77 +1,51 @@
 import * as _ from 'lodash';
+import { PrometheusHealthHandler } from '@console/plugin-sdk';
 import { HealthState } from '@console/shared/src/components/dashboard/status-card/states';
-import { PrometheusResponse } from '@console/internal/components/graphs';
-import { FirehoseResult } from '@console/internal/components/utils';
 import { getGaugeValue } from '../../utils';
 
-const NooBaaStatus = [
-  {
-    state: HealthState.ERROR,
-    message: 'MCG is not running',
-  },
-  {
+const NooBaaStatus = {
+  '0': { state: HealthState.OK },
+  '1': {
     state: HealthState.ERROR,
     message: 'All resources are unhealthy',
   },
-  {
+  '2': {
     state: HealthState.WARNING,
     message: 'Object Bucket has an issue',
   },
-  {
+  '3': {
     state: HealthState.ERROR,
     message: 'Many buckets have issues',
   },
-  {
+  '4': {
     state: HealthState.WARNING,
     message: 'Some buckets have issues',
   },
-];
+};
 
-export const getNooBaaState: GetObjectServiceStatus = (
-  prometheusResponses = [],
-  hasLoadError,
-  isLoading,
-  k8sResponse,
-) => {
-  const [buckets, unhealthyBuckets, pools, unhealthyPools] = prometheusResponses.map((r) =>
-    getGaugeValue(r),
-  );
-  const noobaaPhase = _.get(k8sResponse, 'data[0].status.phase');
-  const unhealthyBucketsRatio = unhealthyBuckets / buckets;
-  const noData = !(buckets && unhealthyBuckets && pools && unhealthyPools && noobaaPhase);
+export const getNooBaaState: PrometheusHealthHandler = (responses, noobaa) => {
+  const { response, error } = responses[0];
+  const noobaaLoaded = noobaa?.loaded;
+  const noobaaLoadError = noobaa?.loadError;
+  const noobaaPhase: string = _.get(noobaa, 'data[0].status.phase');
+  const statusIndex: string = getGaugeValue(response);
 
-  if (hasLoadError) {
+  if (error || noobaaLoadError) {
```

Comment:
use if-else and prioritize this accordingly.

Reply:
We are using return , is that not suffice ?